File: task-03/Python/Sub-Task4.py
def generate_diamond(n):
    diamond = []
    
    # Top half including the middle line
    for i in range(n):
        line = ' ' * (n - i - 1) + '*' * (2 * i + 1)
        diamond.append(line)
    
    # Bottom half excluding the middle line
    for i in range(n - 2, -1, -1):
        line = ' ' * (n - i - 1) + '*' * (2 * i + 1)
        diamond.append(line)
    
    return "\n".join(diamond)

def main():
    try:
        # Read the number from input.txt
        with open('input.txt', 'r') as file:
            n = int(file.read().strip())
        
        # Generate the diamond pattern
        diamond_pattern = generate_diamond(n)
        
        # Write the diamond pattern to output.txt
        with open('output.txt', 'w') as file:
            file.write(diamond_pattern)
        
        print("Diamond pattern generated successfully.")
    except FileNotFoundError:
        print("Error: 'input.txt' file not found.")
    except ValueError:
        print("Error: Invalid number in 'input.txt'.")
    except Exception as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    main()

File: task-03/Ruby/Sub-Task4.rb
# Function to generate the diamond pattern
def generate_diamond(n)
  diamond = ""

  # Top half including the middle line
  (0...n).each do |i|
    # Append leading spaces
    diamond << ' ' * (n - i - 1)
    # Append asterisks
    diamond << '*' * (2 * i + 1)
    # Append new line
    diamond << "\n"
  end

  # Bottom half excluding the middle line
  (n - 2).downto(0) do |i|
    # Append leading spaces
    diamond << ' ' * (n - i - 1)
    # Append asterisks
    diamond << '*' * (2 * i + 1)
    # Append new line
    diamond << "\n"
  end

  diamond
end

# Define file paths
input_file = 'input.txt'
output_file = 'output.txt'

begin
  # Read the number from input.txt
  content = File.read(input_file).strip
  n = content.to_i

  # Validate the input
  if n > 0
    # Generate the diamond pattern
    diamond_pattern = generate_diamond(n)

    # Write the pattern to output.txt
    File.write(output_file, diamond_pattern)

    puts "Diamond pattern successfully written to #{output_file}."
  else
    puts 'Please enter a positive integer in input.txt.'
  end
rescue Errno::ENOENT => e
  puts "An error occurred: #{e.message}"
rescue IOError => e
  puts "An I/O error occurred: #{e.message}"
rescue ArgumentError => e
  puts "Invalid input: #{e.message}"
end
